From 5e840e6292825fcae90f6750a8f57bc989e28c5f Mon Sep 17 00:00:00 2001
From: Gerd Hoffmann <kraxel@redhat.com>
Date: Tue, 26 Apr 2016 15:39:22 +0200
Subject: [PATCH 4/5] vga: update vga register setup on vbe changes

Call the new vbe_update_vgaregs() function on vbe configuration
changes, to make sure vga registers are up-to-date.

Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
[Backport to qemu-xen-tradition]
Signed-off-by: Andrew Cooper <andrew.cooper3@citrix.com>
---
 hw/vga.c | 1 +
 1 file changed, 1 insertion(+)

Index: xen-4.1.6.1/qemu/hw/vga.c
===================================================================
--- xen-4.1.6.1.orig/qemu/hw/vga.c	2016-05-31 10:09:58.000000000 +0000
+++ xen-4.1.6.1/qemu/hw/vga.c	2016-05-31 10:43:45.000000000 +0000
@@ -724,6 +724,7 @@
                 else
                     s->vbe_start_addr += x * ((s->vbe_regs[VBE_DISPI_INDEX_BPP] + 7) >> 3);
                 s->vbe_start_addr >>= 2;
+                vbe_update_vgaregs(s);
             }
             break;
         default:
